zkchannels-crypto/examples/linear_combination_proofs.rs

```diff
@@ -0,0 +1,285 @@
+use zkchannels_crypto::{
+    pedersen::PedersenParameters,
+    pointcheval_sanders::{KeyPair, PublicKey},
+    proofs::{
+        ChallengeBuilder, CommitmentProof, CommitmentProofBuilder, SignatureProof,
+        SignatureProofBuilder,
+    },
+    Message, Rng,
+};
+use {
+    bls12_381::{G1Projective, Scalar},
+    ff::Field,
+};
+
+/// This example generates three different proofs, each demonstrating a different fundamental piece
+/// of a modular linear combination. That is, each proof demonstrates a piece of the following
+/// equation, where (a, b, c) are public values, (x, y, z) are secret values, and q is the modulus of
+/// the BLS12-381 Scalar field:
+/// a*x + b*y + c === z (mod q)
+fn main() {
```

Comment:
No, it's quite easy. I just wanted to illustrate them separately so it was easier to understand which constraint and check maps to which type of relationship. But since we spell it out explicitly in the docs, I can mash them together here.

Reply:
Interesting. My take is that it's nicer to see the library used in a more elegant way and I find the examples easier to follow if they stem as just applications for the slightly-more-general case. But maybe that is just my math background talking.

Reply:
In particular -- the contraint/check maps would be generalized, too, wouldn't they?